### PR TITLE
Allow changing prefix for the installations

### LINKED ISSUE #1654: Add optional prefix parameter to the template
### Description

Currently the prefix is hardcoded to `/usr/local`, for installing guestagent and containerd.

But that doesn't work, if `/usr` is read-only (like on Container Linux, which uses `/opt`).

So add an optional parameter.

```yaml
# Prefix to use for installing guest agent, and containerd with dependencies (if configured)
# 🟢 Builtin default: /usr/local
prefix: null

```

### PR DESCRIPTION
**Pull request by afbjorklund**
If /usr/local is read-only, you can override the prefix to use something like /opt instead. But it is assumed that /etc is always read-write...

Closes #1654

----

For some reason /opt/bin was not in the PATH of the check, even if it is the PATH of the actual user later on.

Workaround was to check with `test -x` in addition to `command -v`, to see if `nerdctl` has been installed yet.

### CODE REVIEW

File: pkg/cidata/cidata.TEMPLATE.d/boot/25-guestagent-base.sh [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 set -eux
 
+: "${LIMA_CIDATA_PREFIX:=/usr/local}"
+
```
Comment (jandubois) [inline pkg/cidata/cidata.TEMPLATE.d/boot/25-guestagent-base.sh:6]: Why is this needed (also in `40-install-containerd.sh`)?

Lima already provides the default value, so `LIME_CIDATA_PREFIX` should never be empty.

File: examples/default.yaml [hunk 2]
```diff
@@ -158,6 +158,10 @@ caCerts:
   #   YOUR-ORGS-TRUSTED-CA-CERT-HERE
   #   -----END CERTIFICATE-----
 
+# Prefix to use for installing guest agent, and containerd with dependencies (if configured)
+# 🟢 Builtin default: /usr/local
+prefix: null
```
Comment (jandubois) [inline examples/default.yaml:163]: `prefix` feels a bit too generic to me in the context of `lima.yaml`. Maybe `installPrefix` or even `guestInstallPrefix` would be better?

Comment (AkihiroSuda) [reply examples/default.yaml:163]: What shout "guestAgentInstallPrefix"

Comment (afbjorklund) [reply examples/default.yaml:163]: It is also used for nerdctl-full install prefix

File: pkg/cidata/cidata.TEMPLATE.d/boot/25-guestagent-base.sh (hunk 1 above)
Comment (afbjorklund) [reply pkg/cidata/cidata.TEMPLATE.d/boot/25-guestagent-base.sh:6]: Maybe it was not needed, but there were some weird test failures installing to "bin/nerdctl"

Comment (afbjorklund) [reply pkg/cidata/cidata.TEMPLATE.d/boot/25-guestagent-base.sh:6]: Will revert

File: examples/default.yaml (hunk 2 above)
Comment (afbjorklund) [reply examples/default.yaml]: Trying with `guestInstallPrefix`

File: examples/default.yaml [hunk 3]
```diff
@@ -158,6 +158,10 @@ caCerts:
   #   YOUR-ORGS-TRUSTED-CA-CERT-HERE
   #   -----END CERTIFICATE-----
 
+# Prefix to use for installing guest agent, and containerd with dependencies (if configured)
+# 🟢 Builtin default: /usr/local
+guestInstallPrefix: null
```
Comment (AkihiroSuda) [inline examples/default.yaml:163]: This should be probably in the "FURTHER ADVANCED CONFIGURATION" section

File: pkg/limayaml/limayaml.go [hunk 4]
```diff
@@ -23,6 +23,7 @@ type LimaYAML struct {
 	Video           Video           `yaml:"video,omitempty" json:"video,omitempty"`
 	Provision       []Provision     `yaml:"provision,omitempty" json:"provision,omitempty"`
 	Containerd      Containerd      `yaml:"containerd,omitempty" json:"containerd,omitempty"`
+	Prefix          *string         `yaml:"guestInstallPrefix,omitempty" json:"guestInstallPrefix,omitempty"`
```
Comment (AkihiroSuda) [inline pkg/limayaml/limayaml.go:26]: Please rename the go variable too

File: pkg/cidata/template.go [hunk 5]
```diff
@@ -58,6 +58,7 @@ type TemplateArgs struct {
 	Mounts                          []Mount
 	MountType                       string
 	Disks                           []Disk
+	Prefix                          string
```
Comment (AkihiroSuda) [inline pkg/cidata/template.go:61]: This needs to be renamed too

**Review by AkihiroSuda — APPROVED** [review]
